extensions-core/kinesis-indexing-service/src/main/java/org/apache/druid/indexing/kinesis/KinesisIndexTaskRunner.java

```diff
@@ -128,8 +128,7 @@ protected void possiblyResetDataSourceMetadata(
       for (final StreamPartition<String> streamPartition : assignment) {
         String sequence = currOffsets.get(streamPartition.getPartitionId());
         String earliestSequenceNumber = recordSupplier.getEarliestSequenceNumber(streamPartition);
```

Comment:
can we pass this to the `isOffsetAvailable` method?

Reply:
since its an expensive call.

Reply:
It's no longer being used here. Thanks for catching this!